internal/handlers/totp.go

```diff
@@ -19,12 +19,17 @@ type TOTPVerifierImpl struct {
 }
 
 // Verify verifies TOTPs.
-func (tv *TOTPVerifierImpl) Verify(token, secret string) (bool, error) {
+func (tv *TOTPVerifierImpl) Verify(token, secret, algorithm string) (bool, error) {
+	algo := otp.AlgorithmSHA512
```

Comment:
Won't this cause an issue if a user sets their algo to SHA256?
Seems default 512 is overridden by the default specified in configuration but no case to handle 256?